app/controllers/vendor_sweets_controller.rb

```diff
@@ -0,0 +1,18 @@
+class VendorSweetsController < ApplicationController
+  def create
+    return redirect_to vendor_path(vendor_sweet_params[:vendor_id]), alert: 'Invalid Parameters' unless valid_params?
+
+    VendorSweet.find_or_create_by!(vendor_sweet_params)
```

Comment:
Best to just use `create` here and handle any uniqueness errors with model validations. 
```suggestion
    VendorSweet.create!(vendor_sweet_params)
```